ktor-io/common/test/io/ktor/utils/io/ByteChannelTest.kt

```diff
@@ -194,4 +194,45 @@ class ByteChannelTest {
         }
         assertEquals(channel.totalBytesRead, 0)
     }
+
+    @Test
+    fun testReadByteWithTimeout() = testSuspend {
+        val channel = ByteChannel()
+
+        launch {
+            channel.writeByte(1)
+            delay(1000)
+            channel.writeByte(0)
+            channel.flush()
+        }
+
+        try {
+            withTimeout(500) {
+                channel.readByte()
+            }
```

Comment:
Can you add an assertion that timeout happened

Reply:
Sure